pkg/k8s/apis/cilium.io/v2/lrp_types.go

```diff
@@ -0,0 +1,146 @@
+//  Copyright 2020 Authors of Cilium
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package v2
+
+import (
+	"github.com/cilium/cilium/pkg/policy/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:singular="localredirectpolicy",path="localredirectpolicies",scope="Namespaced",shortName={lrp}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+
+// LocalRedirectPolicy is a Kubernetes Custom Resource that contains a
+// specification to redirect traffic locally within a node.
+type LocalRedirectPolicy struct {
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Specification of the desired behavior of the local redirect policy
+	Spec LocalRedirectPolicySpec `json:"spec,omitempty"`
+
+	// Most recent status of the local redirect policy
+	// Read-only
+	//
+	// +deepequal-gen=false
+	// +kubebuilder:validation:Optional
+	Status LocalRedirectPolicyStatus `json:"status"`
+}
+
+type L3L4Info struct {
+	// ToIP is a destination IP address for traffic to be redirected.
+	//
+	// Example:
+	// Any traffic destined to '169.254.169.254' is redirected.
+	//
+	// +kubebuilder:validation:Optional
+	ToIP string `json:"toIP,omitempty"`
+
+	// ToPort is a destination port for traffic to be redirected.
+	//
+	// Example:
+	// Any traffic destined to port '53' is redirected.
+	//
+	// +kubebuilder:validation:Optional
+	ToPort string `json:"toPort,omitempty"`
+
+	// L4Proto is layer 4 protocol number for traffic to be redirected.
+	//
+	// +kubebuilder:validation:Enum=TCP;UDP
```

Comment:
Can't we also have `ANY` a a protocol?

Reply:
TCP and UDP are the only supported protocols in LB - https://github.com/cilium/cilium/blob/master/pkg/loadbalancer/loadbalancer.go#L172

Reply:
@aditighag that's is true, but why can't the user write `ANY` and we install 2 LB rules (a TCP and a UDP)?